### ISSUE TITLE
Can't change RabbitMQ default port

### ISSUE THREAD

**Issue by sukinsky** [rabbitmq/rabbitmq-server#287]
Upgraded to 3.5.4 and got a problem: nothing helps to change binding port.
I set port as docs said:
`[{rabbit, [    {tcp_listeners, [33333]]}  ]}].`
I used `RABBITMQ_NODE_IP_ADDRESS=33333`

After downgraded to 3.5.3 everything gone ok.

**Comment by michaelklishin** [comment]
I cannot reproduce with `3.5.4` and the following config:

``` erlang
[
 {rabbit, [
           {tcp_listeners, [15683]}
 ]}
].
```

so the issue is environment-specific. `RABBITMQ_NODE_IP_ADDRESS` is for configuring what IP address the node binds to, not port.

**Comment by michaelklishin** [comment]
@sukinsky please post questions to rabbitmq-users.

**Comment by sukinsky** [comment]
Hi @michaelklishin!
I'm using MS Windows Server 2008R2 with `RABBITMQ_BASE=D:\Work\ESB\rabbitmq_base`.
Used your config and run the server, port didn't changed:
![mngmt](https://cloud.githubusercontent.com/assets/6366681/9492241/6d3afd60-4bfe-11e5-8f00-e983f760258e.png)

Made a typo with RABBITMQ_NODE_PORT.

**Comment by michaelklishin** [comment]
@sukinsky what happens if you override `RABBITMQ_CONFIG_FILE`? There were Windows-specific environment variable-related changes in 3.5.4, however, `RABBITMQ_CONFIG_FILE` [default seems correct](https://github.com/rabbitmq/rabbitmq-server/blob/master/scripts/rabbitmq-defaults.bat#L26).

**Comment by sukinsky** [comment]
@michaelklishin, config seems to be ok, I can change ssl and management ports.

Created env variable `RABBITMQ_CONFIG_FILE=D:\Work\ESB\rabbitmq_base\rmq`.
Specially changed config file name. Reinstalled service and port didn't changed:
![mngmt2](https://cloud.githubusercontent.com/assets/6366681/9492704/a7e2f3a6-4c02-11e5-9dc9-6bcc2167d182.png)

**Comment by michaelklishin** [comment]
@sukinsky there is nothing Windows-specific after the config is loaded. I wonder if the config file value you see in the management UI ends up being different from what is actually used, although that is quite unlikely. Have you checked RabbitMQ logs?

**Comment by sukinsky** [comment]
@michaelklishin, made another test with this config:
`[
	{rabbit, 
		[ 
			{tcp_listeners, [35000]},
			{ssl_listeners, [35001]}
		]
	},
	{rabbitmq_management, 
		[
			{listener, [{port, 35002}]}
		]
	}
].`

And here is what I get:
![mngmt3](https://cloud.githubusercontent.com/assets/6366681/9493528/edb066ba-4c08-11e5-9171-4ea6142653c7.png)

If I make a mistake in tcp_listeners port number or smth else - service will not start, so config is loading correct.

Logs are clean:
`
=INFO REPORT==== 26-Aug-2015::15:39:54 ===
Starting RabbitMQ 3.5.4 on Erlang 18.0
Copyright (C) 2007-2015 Pivotal Software, Inc.
Licensed under the MPL.  See http://www.rabbitmq.com/

=INFO REPORT==== 26-Aug-2015::15:39:54 ===
node           : rabbit@srv
home dir       : C:\Windows
config file(s) : d:/Work/ESB/rabbitmq_base/rmq.config
cookie hash    : BizMGvvRTtTkxYoG3yBwAA==
log            : D:/Work/ESB/rabbitmq_base/log/rabbit@srv.log
sasl log       : D:/Work/ESB/rabbitmq_base/log/rabbit@srv-sasl.log
database dir   : d:/Work/ESB/rabbitmq_base/db/rabbit@srv-mnesia

=WARNING REPORT==== 26-Aug-2015::15:39:54 ===
Kernel poll (epoll, kqueue, etc) is disabled. Throughput and CPU utilization may worsen.

=INFO REPORT==== 26-Aug-2015::15:39:55 ===
Memory limit set to 44232MB of 110581MB total.

=INFO REPORT==== 26-Aug-2015::15:39:55 ===
Disk free limit set to 50MB

=INFO REPORT==== 26-Aug-2015::15:39:55 ===
Limiting to approx 8092 file handles (7280 sockets)

=INFO REPORT==== 26-Aug-2015::15:39:55 ===
Priority queues enabled, real BQ is rabbit_variable_queue

=INFO REPORT==== 26-Aug-2015::15:39:55 ===
Management plugin: using rates mode 'basic'

=INFO REPORT==== 26-Aug-2015::15:39:55 ===
msg_store_transient: using rabbit_msg_store_ets_index to provide index

=INFO REPORT==== 26-Aug-2015::15:39:55 ===
msg_store_persistent: using rabbit_msg_store_ets_index to provide index

=INFO REPORT==== 26-Aug-2015::15:39:56 ===
started TCP Listener on [::]:5672

=INFO REPORT==== 26-Aug-2015::15:39:56 ===
started TCP Listener on 0.0.0.0:5672

=INFO REPORT==== 26-Aug-2015::15:39:56 ===
started SSL Listener on [::]:35001

=INFO REPORT==== 26-Aug-2015::15:39:56 ===
started SSL Listener on 0.0.0.0:35001

=INFO REPORT==== 26-Aug-2015::15:39:56 ===
Management plugin started. Port: 35002

=INFO REPORT==== 26-Aug-2015::15:39:56 ===
Statistics database started.

=INFO REPORT==== 26-Aug-2015::15:39:56 ===
Server startup complete; 6 plugins started.
 * rabbitmq_management
 * rabbitmq_web_dispatch
 * rabbitmq_management_agent
 * amqp_client
 * webmachine
 * mochiweb
`

**Comment by michaelklishin** [comment]
What does `rabbitmqctl report` output?

**Comment by sukinsky** [comment]
@michaelklishin here is 3.5.3 with everything same:
![mngmt4](https://cloud.githubusercontent.com/assets/6366681/9493725/85943da2-4c0a-11e5-8fc0-8eb020550467.png)

**Comment by sukinsky** [comment]
3.5.3 report is here https://yadi.sk/i/7_IiV6UCifuF8
3.5.4 https://yadi.sk/i/fwhLuTd-ifuFG

difference:
3.5.4 {tcp_listeners,[{"auto",5672}]},
3.5.3 {tcp_listeners,[35000]},

**Comment by michaelklishin** [comment]
@sukinsky please use gist.github.com.

**Comment by sukinsky** [comment]
@michaelklishin updated my previous post

**Comment by michaelklishin** [comment]
OK, this seems like a genuine bug. Filed #288.

**Comment by michaelklishin** [comment]
@sukinsky I still cannot reproduce this issue. Can you clarify what exactly you do?

 * How was RabbitMQ installed?
 * What path the config file is at
 * How do you make RabbitMQ use that file

**Comment by sukinsky** [comment]
@michaelklishin I used installers `otp_win64_18.0.exe` and `rabbitmq-server-3.5.4`.exe on a clean WS2008R2.

Next I made an env variable `RABBITMQ_BASE=D:\Work\ESB\rabbitmq_base`
So config is here: `D:\Work\ESB\rabbitmq_base\rabbitmq.config`

In that configuration Rabbit finds config file by itself. I can change any settings besides `tcp_listeners`.
You asked me to set env var for config file explicit like  'RABBITMQ_CONFIG_FILE=D:\Work\ESB\rabbitmq_base\rabbitmq.config` but didn't help 


I tried again on my local PC. Installed OTP and Rabbit. Didn't change any env vars, just used the same config. Setting `tcp_listeners` didn't work in 3.5.4, in 3.5.3 -works.

**Comment by michaelklishin** [comment]
Thanks. I cannot reproduce with a config file overridden with `RABBITMQ_CONFIG_FILE`, will try `RABBITMQ_BASE`.

**Comment by michaelklishin** [comment]
@sukinsky is RabbitMQ running as a Windows service?

**Comment by sukinsky** [comment]
@michaelklishin yes

**Comment by michaelklishin** [comment]
@sukinsky does overriding `RABBITMQ_NODE_PORT` make it use the custom port?

**Comment by michaelklishin** [comment]
@sukinsky what does

```
rabbitmqctl.bat eval "os:getenv()."
```

output? (feel free to edit sensitive pairs out)

**Comment by michaelklishin** [comment]
managed to reproduce. Lets take this to #288 which is a lot more focused on what the actual issue is.

**Comment by sukinsky** [comment]
@michaelklishin good, thank you

**Comment by michaelklishin** [comment]
To answer my own question: `RABBITMQ_NODE_PORT` works just like before (but requires `RABBITMQ_NODE_IP_ADDRESS` to be set) and can be used as a workaround.

**Comment by michaelklishin** [comment]
Just leaving a note here for those not watching #288: the issue is fixed in stable and will be in `3.5.5`. @sukinsky if you get in touch with me at `michael` in the RabbitMQ domain, I can send you a build from the tip of the `stable` branch to try out.
